### ISSUE TITLE
Wont record anymore. Port is closed immediately

### ISSUE THREAD

**Issue by MAISTA23** [andrey-leshenko/WaTracker#14]
Worked fine yesterday. Today only this :

![unbenannt1](https://user-images.githubusercontent.com/28226054/38677952-060a14b2-3e60-11e8-821b-4fe846817e62.png)

![38665861-c1060e7e-3e3d-11e8-97e5-c29404c49d60](https://user-images.githubusercontent.com/28226054/38677957-07b56f0a-3e60-11e8-86a6-3abd4d514fa2.png)

Theres at least one other user with the same problem. What happened?

**Comment by johncormack** [comment]
same issue

whatsapp probably changed something in how it work

**Comment by Community1Systems** [comment]
same issue:
Message dropped because the port is closed
Signed in but not recording

**Comment by SpaceTraveller1** [comment]
Same issue

**Comment by kocaa** [comment]
me too

**Comment by SpaceTraveller1** [comment]
Have you guys found any solution or good alternative?

**Comment by erik1o6** [comment]
It still works fo me without an issue. Running windows 10

**Comment by ga-mer** [comment]
This was a very powerful and useful app indeed!
H+lping in any manner?

**Comment by ga-mer** [comment]
erik1o6: is whatsapp updating the servers in all the countries at once?

**Comment by SpaceTraveller1** [comment]
@ga-mer Yeah they probably did something to their systems, but not all at once worldwide, so maybe somewhere this amazing stool is still up and running.

**Comment by MAISTA23** [comment]
Anyone here good at this code? The record seems to still recognize online users. It just wont report to online/timeline. There should be a workaround. I tried a bit but failed ;D

**Comment by kedarac** [comment]
I have workaround for this problem.. Will post as soon as I get home.

**Comment by MAISTA23** [comment]
> I have workaround for this problem.. Will post as soon as I get home.

Nice ;)

**Comment by erik1o6** [comment]
Awesome great job

**Comment by kedarac** [comment]
I guess due to some server side changes in web whatsapp, the 'wa_stream_start' message type never occurs, instead we see a 'wa_logged_in' message. So I have modified the code in record.js file to accommodate the changes. Do note that this fix is in no way a permanent solution and may need a better solution after studying all the changes that have occurred in web whatsapp.

Attached record.js file, replace the one in the folder by this and reload the extension in chrome. Will send a pull request once I get the time to make some enhancements to this awesome tracker.

[record.zip](https://github.com/andrey-leshenko/WaTracker/files/1916130/record.zip)

**Comment by johncormack** [comment]
btw the app still works despite the warning in the readme file about google dropping support for chrome apps

was this canned?

**Comment by MAISTA23** [comment]
> I guess due to some server side changes in web whatsapp, the 'wa_stream_start' message type never occurs, instead we see a 'wa_logged_in' message. So I have modified the code in record.js file to accommodate the changes. Do note that this fix is in no way a permanent solution and may need a better solution after studying all the changes that have occurred in web whatsapp.

> Attached record.js file, replace the one in the folder by this and reload the extension in chrome. Will send a pull request once I get the time to make some enhancements to this awesome tracker.

> record.zip

Thanks alot. Do you have any workaround for the tracker only recording contacts with an active conversation tab? writing something and not sending it doesnt work for me
NVM, got it. You have to click the contacts once to get recognized

**Comment by Community1Systems** [comment]
Wow
WaTracker is back!
Thanks a lot for that

**Comment by Fueidi2** [comment]
WOW !!!
Not even is it working again, you also fixed the stream_end error, so you had to restart.

@kedaranmol you are fucking awesome

**Comment by alekssaff** [comment]
Doesnt work. 'Message dropped because the port is closed' Ubuntu linux chrome
I had to insert lines
`			if (!port)
				port = chrome.runtime.connect({name: 'presenceUpdates'});`
in record.js in line 58

And it works ))

**Comment by lore132** [comment]
> NVM, got it. You have to click the contacts once to get recognized

That's not working for me. After a few minutes it doesn't recognize online activities as long as you don't have an open conversation with the contact or, at least, open the empty chat on smartphone and laptop.

Any solution for that? Did WhatsApp change things here?

**Comment by MAISTA23** [comment]
On 2nd may whatsapp changed something. no tracking software is working right now
